src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,593 @@
+            this.tableServiceClient = new TableServiceClient(this.settings.StorageConnectionString);
+            this.partitionTable = tableServiceClient.GetTableClient(this.partitionTableName);
+            this.tableLeaseManager = new TableLeaseManager(this.partitionTable, this.service, this.settings, this.storageAccountName);
+        }
+
+        
+        /// <summary>
+        /// This method create a new instance of the class TableLeaseManager that represents the worker. 
+        /// And then start the loop that the worker keeps operating on the table. 
+        /// </summary>
+        async Task IPartitionManager.StartAsync()
+        {
+            await Task.Factory.StartNew(() => this.PartitionManagerLoop(this.partitionManagerCancellationSource.Token));
+            this.settings.Logger.PartitionManagerInfo(
+                this.storageAccountName,
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "", //Empty string as it does not target any particular partition, but rather only initiates the partition manager.
+                $"Worker {this.settings.WorkerId} starts working on acquiring and balancing leases.");
+        }
+
+        async Task PartitionManagerLoop(CancellationToken token)
+        {
+            while (!token.IsCancellationRequested)
+            {
+                TimeSpan timeToSleep = this.settings.LeaseRenewInterval;
+                try
+                {
+                    ReadTableReponse response = await this.tableLeaseManager.ReadAndWriteTable();
+                    if (response.WorkOnRelease || response.WaitForPartition)
+                    {
+                        timeToSleep = TimeSpan.FromSeconds(1);
+                    }
+                }
+                catch
+                {
+                    // if the worker failed to update the table, re-read the table immediately without waiting.
+                    timeToSleep = TimeSpan.FromSeconds(0);
+                }
+                await Task.Delay(timeToSleep, token);
+            }
+        }
+
+        /// <summary>
+        /// This method will stop the partition manager. It first stops the task ReadAndWriteTable().
+        /// And then start the Task ShutDown() until all the leases in the worker is drained. 
+        /// </summary>
+        async Task IPartitionManager.StopAsync()
+        {
+            this.partitionManagerCancellationSource.Cancel();
+            this.settings.Logger.PartitionManagerInfo(
+                this.storageAccountName,
+                this.settings.TaskHubName,
+                this.settings.WorkerId,
+                "",
+                $"Worker {this.settings.WorkerId} strats draining all ownership leases.");
+
+            bool shouldRetry = false;
+            //Shutting down is to drain all the current leases and then release them.
+            //Thus the worker checks table every one sec to ensure timely updates.
```

Comment:
nit: let's fully spell out "second" :)
```suggestion
            //Thus the worker checks table every 1 second to ensure timely updates.
```